File: src/NubipApp.Models/Departament.cs
using System.Collections.Generic;

namespace NubipApp.Models
{
    public class Departament
    {
        public int DepartamentId { get; set; }
        public string Title { get; set; }
        public string Dean { get; set; }
        public int FacultyId { get; set; }
        public string Email { get; set; }

        public Faculty Faculty { get; set; }

        public ICollection<Teacher> Teachers { get; set; }
        public ICollection<Curriculum> Curriculums { get; set; }
    }
}

File: src/models/departament.ts
import mongoose from "mongoose";

let Schema = mongoose.Schema;

let DepartamentSchema = new Schema({
	departamentId: Number,
	title: String,
	dean: String,
	facultyId: [{
		type: Schema.Types.ObjectId, ref: 'Faculty'
	}],
	email: String
});

let Departament = mongoose.model('Departament', DepartamentSchema);

export default Departament;